unified_planning/engines/compilers/ma_disjunctive_conditions_remover.py

```diff
@@ -0,0 +1,371 @@
+                    for i, cl in a.conditions.items():
+                        interval_list.append(i)
+                        new_cond = dnf.get_dnf_expression(
+                            env.expression_manager.And(cl)
+                        )
+                        if new_cond.is_or():
+                            conditions.append(new_cond.args)
+                        else:
+                            conditions.append([new_cond])
+                    conditions_tuple = cast(
+                        Tuple[List[FNode], ...], product(*conditions)
+                    )
+                    for cond_list in conditions_tuple:
+                        nda = self._ma_create_new_durative_action_with_given_conds_at_given_times(
+                            new_ag, interval_list, cond_list, a, dnf
+                        )
+                        if nda is not None:
+                            new_to_old[nda] = a
+                            new_ag.add_action(nda)
+                else:
+                    raise NotImplementedError
+            new_problem.add_agent(new_ag)
+
+            # Meaningful action is the list of the actions that modify fluents that are not added
+            # just to remove the disjunction from goals
+            ag_actions: List["up.model.Action"] = new_ag.actions[:]
+            meaningful_actions.extend(ag_actions)
+
+            self._ma_goals_without_disjunctions_adding_new_elements(
+                dnf,
+                new_problem,
+                new_ag,
+                new_to_old,
+                new_fluents,
+                problem.goals,
+            )
+
+        # Every meaningful action must set to False every new fluent added.
+        # For the DurativeActions this must happen every time the action modifies something
+        em = env.expression_manager
+        # new_effects is the List of effects that must be added to every meaningful action
+        new_effects: List["up.model.Effect"] = [
+            up.model.Effect(em.FluentExp(f), em.FALSE(), em.TRUE()) for f in new_fluents
+        ]
+        for a in meaningful_actions:
+            # Since we modify the action that is a key in the Dict, we must update the mapping
+            old_action = new_to_old.pop(a)
+            if isinstance(a, InstantaneousAction):
+                for e in new_effects:
+                    a._add_effect_instance(e)
+            elif isinstance(a, DurativeAction):
+                for tim in a.effects:
+                    for e in new_effects:
+                        a._add_effect_instance(tim, e)
+            else:
+                raise NotImplementedError
+            new_to_old[a] = old_action
+
+        return CompilerResult(
+            new_problem, partial(replace_action, map=new_to_old), self.name
+        )
+
+    # "def _ma_goals_without_disjunctions_adding_new_elements" not yet adapted to multi-agent case!
```

Comment:
The comment here says: "not yet adapted to multi-agent case!"

The method contains commented code!